pkg/testing/fixture.go

```diff
@@ -239,28 +255,57 @@ func (f *Fixture) Run(ctx context.Context, states ...State) error {
 				return fmt.Errorf("elastic-agent client received unexpected error: %w", err)
 			}
 		case state := <-stateCh:
-			cfg, cont, err := sm.next(state)
-			if err != nil {
-				killProc()
-				return fmt.Errorf("state management failed with unexpected error: %w", err)
-			}
-			if !cont {
-				if !stopping {
-					// trigger the stop
-					stopping = true
-					_ = proc.Stop()
-				}
-			} else if cfg != "" {
-				err := performConfigure(ctx, c, cfg, 3*time.Second)
+			if sm != nil {
+				cfg, cont, err := sm.next(state)
 				if err != nil {
 					killProc()
-					return err
+					return fmt.Errorf("state management failed with unexpected error: %w", err)
+				}
+				if !cont {
+					if !stopping {
+						// trigger the stop
+						stopping = true
+						_ = proc.Stop()
+					}
+				} else if cfg != "" {
+					err := performConfigure(ctx, c, cfg, 3*time.Second)
+					if err != nil {
+						killProc()
+						return err
+					}
 				}
 			}
 		}
 	}
 }
 
+// Exec provides a way of performing subcommand on the prepared Elastic Agent binary.
+func (f *Fixture) Exec(ctx context.Context, args []string, opts ...process.CmdOption) ([]byte, error) {
```

Comment:
Nitpick: maybe we should leave stdout and stderr separate to make output processing and assertions easier.